### PR TITLE
Uses tiledb_domain_has_dimension

### PR DESCRIPTION
**Pull request by antalakas**

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -255,6 +255,15 @@ func (q *Query) SetBuffer(attributeOrDimension string, buffer interface{}) (*uin
 					err)
 			}
 
+			hasDim, err := domain.HasDimension(attributeOrDimension)
```
Comment (Shelnutt2) [inline query.go:258]: This should be above, the idea is to not call `schema.AttributeFromName` if its a dimension. We should check if its a dim or a attr before trying to fetch the object. This apply everywhere.

Comment (antalakas) [reply query.go:258]: According to TileDB export comments: `tiledb_domain_has_dimension` : `Checks whether the domain has a dimension of the given name.` Will it work with attribute names?

Comment (ihnorton) [reply query.go:258]: There is also `tiledb_array_schema_has_attribute`.

File: query.go [hunk 2]
```diff
@@ -227,27 +227,31 @@ func (q *Query) SetBuffer(attributeOrDimension string, buffer interface{}) (*uin
...
 		}
 	} else {
-		schemaAttribute, err := schema.AttributeFromName(attributeOrDimension)
+		hasDim, err := domain.HasDimension(attributeOrDimension)
 		if err != nil {
+			return nil, err
+		}
+
+		if hasDim {
 			domain, err := schema.Domain()
```
Comment (Shelnutt2) [inline query.go:255]: you can remove domain here, you fetched it above

**Pull request merged**